File: src/main/java/client/messages/broadcast/types/NoticeWithoutPrefixMessage.java
package client.messages.broadcast.types;

import client.messages.broadcast.AbstractBroadcastMessage;
import client.messages.broadcast.BroadcastMessageType;
import util.packet.PacketWriter;

/**
 * Color: Blue
 * Prefix:
 */
public class NoticeWithoutPrefixMessage extends AbstractBroadcastMessage {

    private final String message;

    public NoticeWithoutPrefixMessage(String message) {
        this.message = message;
    }

    @Override
    public BroadcastMessageType getType() {
        return BroadcastMessageType.NOTICE_WITHOUT_PREFIX;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeMapleString(message);
        pw.writeInt(0);
    }
}

File: src/main/java/client/messages/broadcast/types/NoticeWithoutPrefixMessage.kt
package client.messages.broadcast.types

import client.messages.broadcast.AbstractBroadcastMessage
import client.messages.broadcast.BroadcastMessageType
import util.packet.PacketWriter

/**
 * Color: Blue
 * Prefix:
 */
class NoticeWithoutPrefixMessage(private val message: String) : AbstractBroadcastMessage() {

    override val type: BroadcastMessageType get() = BroadcastMessageType.NOTICE_WITHOUT_PREFIX

    override fun encodeData(pw: PacketWriter) {
        pw.writeMapleString(message)
        pw.writeInt(0)
    }
}